src/Microsoft.Azure.SignalR/HubHost/ServiceHubDispatcher.cs

```diff
@@ -76,15 +76,24 @@ public async Task ShutdownAsync()
                 return;
             }
 
-            using CancellationTokenSource source = new CancellationTokenSource();
-            source.CancelAfter(_options.GracefulShutdown.Timeout);
+            try
+            {
+                var source = new CancellationTokenSource(_options.GracefulShutdown.Timeout);
 
-            await Task.WhenAny(
-                OfflineAndWaitForCompletedAsync(_options.GracefulShutdown.Mode),
-                Task.Delay(TimeSpan.FromHours(1), source.Token)
-            );
+                await Task.WhenAny(
+                    OfflineAndWaitForCompletedAsync(_options.GracefulShutdown.Mode),
+                    Task.Delay(TimeSpan.FromDays(1), source.Token)
```

Comment:
`Timeout.InfiniteTimeSpan`, there is also an `OrTimeout` extension method to leverage